x/ecocredit/core/genesis_test.go

```diff
@@ -1,173 +1,151 @@
-			}
-		],
-		"regen.ecocredit.v1.Project":[
-			{
-				"id":"P01",
-				"admin":"gPFuHL7Hn+uVYD6XOR00du3C/Xg=",
-				"class_key":"1",
-				"project_jurisdiction":"AQ"
-			},
-			{
-				"id":"P02",
-				"admin":"CHkV2Tv6A7RXPJYTivVklbxXWP8=",
-				"class_key":"2",
-				"project_jurisdiction":"AQ",
-				"metadata":"project metadata"
-			}
-		],
-		"regen.ecocredit.v1.CreditType": [
-			{
-				"name":"carbon",
-				"abbreviation": "C",
-				"unit":"metric ton C02 equivalent",
-				"precision":6
-			},
-			{
-				"name":"biodiversity",
-				"abbreviation": "BIO",
-				"unit":"acres",
-				"precision":6
-			}
-		]
-	}`
+	ormCtx := ormtable.WrapContextDefault(ormtest.NewMemoryBackend())
+	modDB, err := ormdb.NewModuleDB(&ecocredit.ModuleSchema, ormdb.ModuleDBOptions{})
+	require.NoError(t, err)
+	ss, err := api.NewStateStore(modDB)
+
+	require.NoError(t, ss.CreditTypeTable().Insert(ormCtx, &api.CreditType{
+		Abbreviation: "BIO",
+		Name:         "biodiversity",
+		Unit:         "acres",
+		Precision:    6,
+	}))
+
+	require.NoError(t, ss.BatchBalanceTable().Insert(ormCtx,
+		&api.BatchBalance{BatchKey: 1, Address: sdk.AccAddress("addr1"), Tradable: "90.003", Retired: "9.997"}))
+
+	batches := []*api.Batch{
+		{Issuer: sdk.AccAddress("addr2"), ProjectKey: 1, Denom: "BIO01-00000000-00000000-001", StartDate: &timestamppb.Timestamp{Seconds: 100}, EndDate: &timestamppb.Timestamp{Seconds: 101}},
+		{Issuer: sdk.AccAddress("addr3"), ProjectKey: 1, Denom: "BIO02-00000000-00000000-001", StartDate: &timestamppb.Timestamp{Seconds: 100}, EndDate: &timestamppb.Timestamp{Seconds: 101}},
+	}
+	for _, b := range batches {
+		require.NoError(t, ss.BatchTable().Insert(ormCtx, b))
+	}
+
+	require.NoError(t, ss.BatchSupplyTable().Insert(ormCtx, &api.BatchSupply{BatchKey: 1, TradableAmount: "90.003", RetiredAmount: "9.997"}))
+
+	classes := []*api.Class{
+		{Id: "BIO001", Admin: sdk.AccAddress("addr4"), CreditTypeAbbrev: "BIO"},
+		{Id: "BIO002", Admin: sdk.AccAddress("addr5"), CreditTypeAbbrev: "BIO"},
+	}
+	for _, c := range classes {
+		require.NoError(t, ss.ClassTable().Insert(ormCtx, c))
+	}
+
+	projects := []*api.Project{
+		{Id: "P01", Admin: sdk.AccAddress("addr6"), ClassKey: 1, ProjectJurisdiction: "AQ", Metadata: "meta"},
+		{Id: "P02", Admin: sdk.AccAddress("addr7"), ClassKey: 2, ProjectJurisdiction: "AQ", Metadata: "meta"},
+	}
+	for _, p := range projects {
+		require.NoError(t, ss.ProjectTable().Insert(ormCtx, p))
+	}
+
+	target := ormjson.NewRawMessageTarget()
+	require.NoError(t, modDB.ExportJSON(ormCtx, target))
+	genesisJson, err := target.JSON()
+	require.NoError(t, err)
 
 	params := core.Params{
 		AllowlistEnabled: true,
 	}
-	err := core.ValidateGenesis(json.RawMessage(x), params)
+	err = core.ValidateGenesis(genesisJson, params)
 	require.NoError(t, err)
 }
 
 func TestGenesisValidate(t *testing.T) {
 	defaultParams := core.DefaultParams()
-
+	dummyAddr := sdk.AccAddress("foobar")
+	dummyAddr2 := sdk.AccAddress("fooBarBaz")
```

Comment:
No need for the `dummy` prefix. Might be nice to use `addr1` and `addr2`.

Reply:
https://github.com/regen-network/regen-ledger/pull/1055/commits/4e7ead11521ba222992d4ea52d21f3867d6fe814